### DISCUSSION TITLE
Ability to (Deep)link into LobeChat with initial prompt and other parameters

### DISCUSSION THREAD

**Discussion by dhLubv7** [lobehub/lobehub#2766] [Ideas]
Hello,
I am fairly new to LobeChat and I have a rough idea where I would like to (deep)link into LobeChat with an initial prompt.

Imagine you are in an productivity app like Raycast, Alfred or Notion and you could write a prompt and then be directly forwarded to a new conversation in LobeChat where your initial prompt is automatically converted into a new message.

From my understanding this should be fairly easy to implement, as the prompt could just be transmitted as URL parameter (e.g. ?query=....). It would be a nice feature for all those power app users (like Raycast).

Just a rough idea though, what do you think?

**Comment by arvinxx** [comment]
I think it's a good idea. There's some stuff we need to discuss. 

The query is send to inbox session or which Agent or show up a spotlight?

**Comment by tcmonster** [comment]
sounds like some kinda API feature.  how about responses in Raycast directly

**Comment by dhLubv7** [comment]
I am personally a fan of an MVP approach, meaning that we start with a very simple version and then see if there are people using it and asking for more.

If we would want to display responses directly in Raycast and then maybe even respond to them it gets immediately way more complex. Therefore I would propose we start simple:

**Concept for a simple MVP:**

In LobeChat:

I would suggest that on page load LobeChat simply checks if there are URL Parameters presents.

- `query`: The prompt the user wants to initiate (v1)
- `model`: The model to use (v2)
- `agent`: The agent to use (v3)

If `query` is present the page simply handles the value like it would handle user input and post the first message. 

In Raycast:

Remember that Raycast is just one use case here. With parameters like the ones mentioned above you could trigger LobeChat from anywhere really. For Raycast a very simple extension should do the trick, that in the end simply opens an URL via 
[Action.OpenInBrowser](https://developers.raycast.com/api-reference/user-interface/actions#action.openinbrowser).

Example:

1. The user opens Raycast
2. The user toggles the extension "LobeChat"
3. The user then just writes a prompt like `"How many countries are on our planet?"`
4. The user hits `enter`
5. The default browser is opened with the configured LobeChat Url (within the extension settings in Raycast). The Url is something like: `https://[lobe-chat-url]?query=How%20many%20countries%20are%20on%20our%20planet?`
6. In LobeChat the query is extracted and an initial prompt is send

What do you think?

**Comment by tcmonster** [comment]
one question, is there any way to be compatible with the PWA app?

**Comment by DBFritz** [comment]
I recently discover that using `https://[lobe-chat-url]/chat?message=[initial-prompt]` does the trick.
